### PR TITLE
React v17 support

### PR DESCRIPTION
**Pull request by stephensauceda**
#### What does this PR do?

Bumps dependencies to support React 17. This is package blocks some of our other apps and services from being able to upgrade due to peerDependencies (namely oxygen and the auth migration).

#### How can this change be manually tested?

I ran the tests with `yarn test`. It appears two tests are failing. But they appear to be the same two tests that fail on `master`. Not sure how you guys are rolling with this. Are the tests known failures? 

>SUMMARY:
>✔ 65 tests completed
>✖ 2 tests failed
>
>FAILED TESTS:
 > Bling
 >   ✖ reflects outOfPage props to adSlot
 >     Chrome 110.0.0 (Mac OS X 10.15.7)
 >   Error: Uncaught TypeError: Cannot read properties of undefined (reading 'INTERSTITIAL') (webpack:///src/Bling.js:9:23639 <- test/Bling.spec.js:21225)
>
>    ✖ sets slotSize to 0,0 on foldCheck of 'fluid' or ['fluid']
>      Chrome 110.0.0 (Mac OS X 10.15.7)
>    Error: Timeout of 2000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.

To test the ad loading stuff:
1. pull down this branch. 
2. In the `react-gpt` directory, `yarn` and `yarn build` then, run `npm link`
3. Open up oxygen in your terminal
4. Switch to the `s/react17` branch
5. Run `npm link @atmedia/react-gpt`
6. Load up the sites as you normally would locally.

#### Asana Card

https://app.asana.com/0/1201709980437746/1203717210397677/f

#### Questions or concerns about this change?

~When I bump the React dependencies to v17, I get a peerDependency warning for [Radium](https://www.npmjs.com/package/radium). This is a bummer because it looks like it's a package that is no longer [supported by the maintainers.](https://github.com/formidablelabs/radium) They encourage forks, so... I guess we do that?~

#### Deployment Notes

~Need to resolve the peerDependency on Radium and React v16~ Radium has React 17 support in 0.26.2

---

#### PR Readiness

* [ ] Changes have been validated in local dev

#### Code Readiness

* [ ] Code follows [AT Media styles and conventions](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/RCwBYjgA4B)
* [ ] Code has been [auto-formatted and is free of lint errors](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/BXnBJVigtW)
* [ ] Unrelated code changes are not present
* [ ] Relevant inline documentation or READMEs have been updated
* [ ] Application and browser logs reviewed for runtime warnings and errors
* [ ] Correct level of [logging](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/gwTZjrJhOy) added to code

#### Automated Testing

* [ ] [Unit tests](https://apartmenttherapy.slite.com/app/channels/R54jSBx0aG/notes/k6Uah8Iquw) are written at the appropriate level given the code’s risk-level
* [ ] Existing tests still pass
* [ ] No unnecessary test runner output, e.g. debugging or warnings are not present in Jest or Unittest output

#### Other Considerations

* [ ] Relevant documentation updated in Slite

Also consider this PR's impact on architecture and security; AMP, Apple News, RSS; SEO, GA, performance and accessibility.

Please see [Post Merge](https://apartmenttherapy.slite.com/app/docs/88ULglsAB9) for additional responsibilities after this PR is merged.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -78,10 +85,9 @@
         "prop-types": "^15.5.10",
         "querystring": "^0.2.0",
         "radium": "^0.25.1",
-        "react": "^16.0.0",
-        "react-addons-test-utils": "^15.0.1",
```
Comment (stephensauceda) [inline package.json:82]: It looks like you guys have already migrated to using `react-dom/test-utils` so this doesn't appear to be needed anymore

File: package.json [hunk 2]
```diff
@@ -92,29 +98,24 @@
...
-        "react-dom": "^15.0.1 || ^16.0.0"
+        "react": "^15.0.1 || ^16.0.0 || ^17.0.2",
+        "react-dom": "^15.0.1 || ^16.0.0 || ^17.0.2"
     },
     "scripts": {
         "commit": "git-cz",
         "changelog": "conventional-changelog -p angular -i CHANGELOG.md -s",
         "build": "npm run clean && npm run compile",
         "build:es": "BABEL_ENV=es babel --copy-files ./src -d es",
-        "build:umd":
```
Comment (stephensauceda) [inline package.json:103]: Prettier got a hold of some of this. The only relevant changes are the deps above.

File: package.json [hunk 3]
```diff
@@ -1,6 +1,6 @@
 {
     "name": "@atmedia/react-gpt",
-    "version": "2.1.6",
+    "version": "2.1.7",
```
Comment (stephensauceda) [inline package.json]: I just bumped this because 🤷 If we're being semantic, it's probably a major version change.

File: src/Bling.js [hunk 4]
```diff
@@ -547,13 +551,13 @@ class Bling extends Component {
...
 
         const inViewport = Bling._adManager.isInViewport(
-            ReactDOM.findDOMNode(this),
+            this.wrapper.current,
             slotSize,
             this.viewableThreshold
         );
 
         const isHidden = () => {
-            const el = ReactDOM.findDOMNode(this);
```
Comment (stephensauceda) [inline src/Bling.js:556]: `findDOMNode` [no longer behaves the way it used to](https://reactjs.org/docs/strict-mode.html#warning-about-deprecated-finddomnode-usage). This looked like the only change needed to get ads rendering. Whether they are rendering _correctly_ I'll leave to you guys ;) at least to tell me that they're broken.

**Comment by Engywook2005** [comment]
@stephensauceda I can have a look into the failing unit tests. The first of the two is almost certainly from our having made some tweaks to the out of page ads to accommodate the mobile interstitial, and can most likely be fixed by updating the test.

Will need to get back to you on the second test which is timing out. I wonder if there's something about this under the `issues` tab under the main repo.

**Pull request merged**